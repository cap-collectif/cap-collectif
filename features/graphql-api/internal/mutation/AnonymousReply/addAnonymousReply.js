/* eslint-env jest */
import '../../../_setup';

const AddAnonymousReplyMutation = /* GraphQL*/ `
    mutation AddAnonymousReplyMutation($input: AddAnonymousReplyInput!) {
        addAnonymousReply(input: $input) {
            reply {
                __typename
                id
            }
            questionnaire {
                id
            }
            participantToken
            errorCode
        }
    }
`;

const input = {
  "questionnaireId": "UXVlc3Rpb25uYWlyZTpxdWVzdGlvbm5haXJlQW5vbnltb3Vz",
  "responses": [
    {
      "value": "abc",
      "question": "UXVlc3Rpb246MTQwMw=="
    }
  ]
};

const propertyMatchers = {
  addAnonymousReply: {
    reply: {
      id: expect.any(String)
    },
    participantToken: expect.any(String)
  },
}

describe('mutations.addAnonymousReply', () => {
  it('should create an anonymous reply', async () => {
    const response = await graphql(
      AddAnonymousReplyMutation,
      {
        input,
      },
      'internal'
    );

    expect(response).toMatchSnapshot(propertyMatchers);
  });
});
