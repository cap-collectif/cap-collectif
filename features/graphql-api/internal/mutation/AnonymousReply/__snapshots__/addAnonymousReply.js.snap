// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mutations.addAnonymousReply should create an anonymous reply 1`] = `
Object {
  "addAnonymousReply": Object {
    "errorCode": null,
    "participantToken": Any<String>,
    "questionnaire": Object {
      "id": "UXVlc3Rpb25uYWlyZTpxdWVzdGlvbm5haXJlQW5vbnltb3Vz",
    },
    "reply": Object {
      "__typename": "Reply",
      "id": Any<String>,
    },
  },
}
`;
